numpy/linalg/_linalg.py

```diff
@@ -2151,6 +2172,23 @@ def pinv(a, rcond=1e-15, hermitian=False):
 
     """
     a, wrap = _makearray(a)
+    if rcond is None:
+        if rtol is _NoValue:
+            rcond = 1e-15
+        elif rtol is None:
+            rcond = max(a.shape[-2:]) * finfo(a.dtype).eps
+        else:
+            rcond = rtol
+    elif rtol is not _NoValue:
+        raise ValueError("`rtol` and `rcond` can't be both set.")
+    else:
+        warnings.warn(
+            "In the future `rcond` (with the default value `1e-15`) will be"
+            "replaced by `rtol` (with the default value `max(M, N) * eps`)",
+            PendingDeprecationWarning,
```

Comment:
I don't think numpy normally uses `PendingDeprecationWarning` because no one will see them unless they've explicitly opted into it in their testing. Better to just go straight to the deprecation.

That said, I think it would be better to add a note here that we should deprecate this in a few versions, once downstream libraries will no longer need to add a numpy version check to support the new keyword and avoid the warning.

Reply:
I think many packages (well, at least astropy and numpy) run their tests with all warnings turned into errors, and that includes `PendingDeprecationWarning` - my tendency would be to keep this *and* add a note.

Reply:
Or did you mean to avoid people getting hit by warnings in their test suites? I.e., add a `DeprecationWarning` when the minimum supported numpy version is 2.0? That makes sense...

Reply:
Yes exactly. I think in general we should be ok with some short-term API warts so downstream doesn't need to maintain a bunch of numpy compat code unless we really need to change something immediately for some
reason.

Reply:
Removed warning and added a note.